scipy/interpolate/_ndbspline.py

```diff
@@ -180,6 +180,10 @@ def __call__(self, xi, *, nu=None, extrapolate=None):
         if xi_shape[-1] != ndim:
             raise ValueError(f"Shapes: xi.shape={xi_shape} and ndim={ndim}")
 
+        if any(nu > np.asarray(self.k, dtype=np.int64) + 1):
+            return np.zeros(xi_shape[:-1] + self._c.shape[ndim:],
+                            dtype=self._c.dtype)
+
```

Comment:
This change is not needed now, after I updated `_deBoor_D` to return `0`s when `m > k + 1`. See the diff of `scipy/interpolate/src/__fitpack.cc`. We can remove this if @ev-br agrees.

Reply:
Sure. Will do once I come back. Will remove this Pythonic change as well. That should do the job I hope.